hazelcast/src/main/java/com/hazelcast/util/function/FunctionEx.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.util.function;
+
+import com.hazelcast.util.ExceptionUtil;
+
+import java.io.Serializable;
+import java.util.function.Function;
+
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * {@code Serializable} variant of {@link Function java.util.function.Function}
+ * which declares checked exception.
+ */
+@FunctionalInterface
+public interface FunctionEx<T, R> extends Function<T, R>, Serializable {
```

Comment:
When this gets merged, we can remove `c.h.core.IFunction` and use `FunctionEx`.

Reply:
Ok, I'll trim the PR and keep the ones we actually would need, possibly substituting `IFunction` in the process.